packages/utils-tests-helpers/src/hint-runner.ts

```diff
@@ -75,16 +75,37 @@ const validateResults = (t: GenericTestContext<Context<any>>, results: Problem[]
         return t.fail(`Result count is ${results.length}, should be ${reports.length}`);
     }
 
-    return reports.forEach((report, index: number) => {
-        if (server.port) {
-            t.is(results[index].message, report.message.replace(localhostRegex, `$1://localhost:${server.port}/`), `Different message`);
-        } else {
-            t.is(results[index].message, report.message, `Different message`);
-        }
-        if (report.position) {
-            t.is(results[index].location.column, report.position.column, `Different column`);
-            t.is(results[index].location.line, report.position.line, `Different line`);
+    if (server.port) {
+        reports.forEach((report) => {
+            report.message = report.message.replace(localhostRegex, `$1://localhost:${server.port}/`);
+        });
+    }
+
+    results.forEach((result) => {
+        const { message } = result;
+        let index = 0;
+        const reportsCopy = reports.slice(0);
```

Comment:
The copy needs to be created outside of `results.forEach`. Otherwise this keeps looking up against the original `reports` on each iteration (effectively cancelling out the removal).